### PR TITLE
Add scope parameter to /api/search/tags

### LINKED ISSUE #2233: [Trace Explorer] Autocompletion suggests fields `span.name` and `.name` but queries fail
**Describe the bug**

Autocompletion suggests `span.name` and `.name` but:
* Autocompletion for the value of `.name` and `span.name` don't work
* Queries return no entry instead of returning the same entries as a query on `name`

Note that the same problem exists for `span.status`

**To Reproduce**

* Ingest traces including server kind spans named `POST /api/orders`
* Navigate to Grafana Explore Tempo and select the TraceQL tab
* Type the following TraceQL queries:
   * `name=`
      * Autocompletion suggest enad of attribute name `name`
      * Autocompletion suggests value `POST /api/orders`
      * Query on `name="POST /api/orders"` returns entries
   * `.name=`
      * Autocompletion suggest end of attribute name `.name`
      * Autocompletion DOESN'T suggest any value
      * Query on `.name="POST /api/orders"` returns NO entry
   * `span.name=`
      * Autocompletion suggest end of attribute name `span.name`
      * Autocompletion DOESN'T suggest any value
      * Query on `span.name="POST /api/orders"` returns NO entry

**Expected behavior**

TraceQL queries on `name="POST /api/orders"`, `.name="POST /api/orders"`, and `span.name="POST /api/orders"` should return the same entries.

**Environment:**
 - Infrastructure: Grafana Cloud as of 2023-03-19

**Screenshot**

* works as expected
image

* fails
image

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
This PR

- adds a `?scope` parameter to the `/api/search/tags` endpoint. This parameter can currently take 3 values (resource, span, and intrinsic) as well as no value which defaults to the old behavior of returning both span and resource tags.
- removes attributes that are non-existent in traceql such as root.name and root.service.name.
- Adds a new endpoint `/api/v2/search/tags` that returns all tags organized by scope.

~Note that this PR is blocked by #2244. Marking this as draft until that is merged.~

**Which issue(s) this PR fixes**:
Fixes #2233 

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/tempo/api_docs/_index.md [hunk 1]
```diff
@@ -247,10 +247,11 @@ $ curl -G -s http://localhost:3200/api/search --data-urlencode 'tags=service.nam
 Ingester configuration `complete_block_timeout` affects how long tags are available for search.
 
 This endpoint retrieves all discovered tag names that can be used in search.  The endpoint is available in the query frontend service in
-a microservices deployment, or the Tempo endpoint in a monolithic mode deployment.
+a microservices deployment, or the Tempo endpoint in a monolithic mode deployment. The tags endpoint takes a scope that controls the kinds 
+of tags or attributes returned. If nothing is provided the endpoint will return all resource and span tags.
```
Comment (knylander-grafana) [inline docs/sources/tempo/api_docs/_index.md:251]:
```suggestion
of tags or attributes returned. If nothing is provided, the endpoint will return all resource and span tags.
```

**Review by knylander-grafana — COMMENTED** [review]
Thank you for adding documentation! Changes look good. One minor suggestion for a missing comma.

**Comment by joe-elliott** [comment]
> Did you consider a new version of the endpoint where you return a list of available scopes for each tag?

I did, but I liked not revving the endpoint b/c then I had to write less code :). If I did create a v2 it would likely return something like this though:

```
{
  "intrinsic": [ "array", "of", "strings"],
  "span": [ "a", "b", "c"],
  "resource": [ "d", "e", "f" ],
  ...
}
```

would you all prefer this?

**Comment by joe-elliott** [comment]
> Yes, that also works! Is it possible for a tag to be in more than one scope?
Yup, but this would likely occur rarely.

**Comment by joe-elliott** [comment]
@adrapereira added a v2 endpoint. Example:

```
$ curl -G -s http://localhost:3200/api/v2/search/tags  | jq
{
  "scopes": [
    {
      "name": "span",
      "tags": [
        "article.count",
        "http.flavor",
        "http.method",
      ]
    },
    {
      "name": "resource",
      "tags": [
        "k6",
        "service.name"
      ]
    },
    {
      "name": "intrinsic",
      "tags": [
        "duration",
        "kind",
        "name",
        "status"
      ]
    }
  ]
}
```

**Review by knylander-grafana — COMMENTED** [review]
Doc addition looks good. Thank you.

File: docs/sources/tempo/api_docs/_index.md [hunk 2]
```diff
@@ -248,10 +249,11 @@ $ curl -G -s http://localhost:3200/api/search --data-urlencode 'tags=service.nam
 Ingester configuration `complete_block_timeout` affects how long tags are available for search.
 
 This endpoint retrieves all discovered tag names that can be used in search.  The endpoint is available in the query frontend service in
-a microservices deployment, or the Tempo endpoint in a monolithic mode deployment.
+a microservices deployment, or the Tempo endpoint in a monolithic mode deployment. The tags endpoint takes a scope that controls the kinds 
+of tags or attributes returned. If nothing is provided, the endpoint will return all resource and span tags.
 
 ```
-GET /api/search/tags
+GET /api/search/tags?scope=<resource|span|intrinsic>
```
Comment (zalegrala) [inline docs/sources/tempo/api_docs/_index.md:256]: Is `scope` still present after the `v2` rev commits that were added?

**Review by zalegrala — COMMENTED** [review]
Looks good to me, just a couple comments.

File: docs/sources/tempo/api_docs/_index.md (hunk 2 above)
Comment (joe-elliott) [reply docs/sources/tempo/api_docs/_index.md:256]: yup

File: pkg/traceql/enum_attributes.go [hunk 3]
```diff
@@ -23,6 +28,19 @@ func (s AttributeScope) String() string {
 	return fmt.Sprintf("att(%d).", s)
 }
 
+func AttributeScopeFromString(s string) AttributeScope {
+	switch s {
+	case "span":
+		return AttributeScopeSpan
+	case "resource":
+		return AttributeScopeResource
+	case "":
```
[the remark this answers is not in the document]
Comment (joe-elliott) [reply pkg/traceql/enum_attributes.go:37]: yup, i like that.

**Review by zalegrala — APPROVED** [review]
Nice work.  Looks good to me.

**Pull request merged**